File: Tests/CorePromise/Infrastructure.m
@import Foundation;
#import "Infrastructure.h"

@implementation XCTestCase (PMKUnhandledErrorHandler)

+ (void)initialize {
    [Injected setUp];
}

- (void)setUp {
    Injected.errorUnhandler = ^(id err) {
        XCTFail("Unexpected unhandled error");
    };
}

@end

File: Tests/CorePromise/Infrastructure.swift
import PromiseKit
import XCTest

var InjectedErrorUnhandler: (Error) -> Void = { _ in XCTFail() }

@objc(PMKInjected) class Injected: NSObject {
    @objc class func setErrorUnhandler(_ newErrorUnhandler: @escaping (NSError) -> Void) {
        InjectedErrorUnhandler = { newErrorUnhandler($0 as NSError) }
    }

    @objc class func setUp() {
        PMKSetUnhandledErrorHandler { err in
            InjectedErrorUnhandler(err)
        }
    }
}
